src/http/server.rs

```diff
@@ -67,16 +70,57 @@ pub async fn start_server(config: Config) -> Result<()> {
   let blueprint = Blueprint::try_from(&config).map_err(CLIError::from)?;
   let http_client = Arc::new(DefaultHttpClient::new(&blueprint.upstream));
   let state = Arc::new(ServerContext::new(blueprint.clone(), http_client));
-  let make_svc = make_service_fn(move |_conn| {
-    let state = Arc::clone(&state);
-    async move { Ok::<_, anyhow::Error>(service_fn(move |req| handle_request(req, state.clone()))) }
-  });
-  let addr = (blueprint.server.hostname, blueprint.server.port).into();
-  let server = hyper::Server::try_bind(&addr).map_err(CLIError::from)?.serve(make_svc);
-  log::info!("🚀 Tailcall launched at [{}]", addr);
-  if blueprint.server.enable_graphiql {
-    log::info!("🌍 Playground: http://{}", addr);
+  let addr: SocketAddr = (blueprint.server.hostname, blueprint.server.port).into();
+
+  if blueprint.server.http.version == HttpVersion::HTTP2
+    && (blueprint.server.http.cert_path.is_none() || blueprint.server.http.key_path.is_none())
+  {
+    return Err(anyhow::anyhow!(CLIError::new("HTTP/2 protocol requires both certificate and key paths. Please provide them using --cert-path and --key-path options.")));
```

Comment:
We could check for cert_path and key_path on the match expression rather than here. In this way, we'd check for HTTP2 version once.